src/Vm.sol

```diff
@@ -3,48 +3,54 @@ pragma solidity >=0.6.2 <0.9.0;
 
 pragma experimental ABIEncoderV2;
 
+// Cheatcodes are marked as view/pure/none using the following rules:
+// 0. A call's observable behaviour includes its return value, logs, reverts and state writes.
+// 1. If you can influence a later call's observable behaviour, you're neither view nor pure (you are modifying some state be it the EVM, interpreter, filesystem, etc),
+// 2. otherwise if you can be influenced by an earlier call, you're `view`,
```

Comment:
Would this be an accurate clarification? For example, `vm.projectRoot` reads some data but is not influenced by earlier calls

```suggestion
// 2. otherwise if you can be influenced by an earlier call, or if reading some state, you're `view`,
```

Reply:
Haha I thought about that but can't ffi change that value?

Reply:
Nevermind, it can't! I'll include your edit.